core-rust/core-api-server/src/core_api/errors.rs

```diff
@@ -128,6 +128,17 @@ pub(crate) fn assert_matching_network<E: ErrorDetails>(
     Ok(())
 }
 
+pub(crate) fn assert_unbounded_endpoints_flag_enabled<E: ErrorDetails>(
+    state: &CoreApiState,
+) -> Result<(), ResponseError<E>> {
+    if !state.flags.enable_unbounded_endpoints {
+        return Err(client_error(
+            "This endpoint is unbounded and `enable_unbounded_endpoints` flag is disabled.",
```

Comment:
Looks nice, although technically it's the response, not the endpoint itself which is unbounded - so I would slightly rather: "This endpoint is disabled as the response from this endpoint is potentially unbounded, and this node is configured with `enable_unbounded_endpoints` false"